client/landing/gutenboarding/components/domain-picker-modal/index.tsx

```diff
@@ -3,47 +3,61 @@
  */
 import * as React from 'react';
 import Modal from 'react-modal';
+import { select } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import DomainPicker, { Props as DomainPickerProps } from '@automattic/domain-picker';
+import { recordEnterModal, recordCloseModal } from '../../lib/analytics';
+import { STORE_KEY } from '../../stores/onboard';
 
 /**
  * Style dependencies
  */
 import './style.scss';
 
-interface Props extends Omit< DomainPickerProps, 'tracksName' > {
+interface Props extends DomainPickerProps {
 	isOpen: boolean;
 	onMoreOptions?: () => void;
 }
 
-const DomainPickerModal: React.FunctionComponent< Props > = ( { isOpen, ...props } ) => {
+const DomainPickerModal: React.FunctionComponent< Props > = ( { isOpen, onClose, ...props } ) => {
 	// This is needed otherwise it throws a warning.
 	Modal.setAppElement( '#wpcom' );
 
-	if ( ! isOpen ) return null;
+	const tracksName = 'DomainPickerModal';
 
-	const handleAfterOpen = () => {
+	const handleOpen = () => {
 		// This fixes modal being shown without
 		// header margin when the modal is opened.
 		window.scrollTo( 0, 0 );
+		recordEnterModal( tracksName );
+	};
+
+	const handleClose = () => {
+		recordCloseModal( tracksName, {
+			selected_domain: select( STORE_KEY ).getSelectedDomain()?.domain_name,
+		} );
+		onClose && onClose();
```

Comment:
NIT: Optional chain can be optional call:

```suggestion
		onClose?.();
```

Reply:
Done.